addons/binding/org.openhab.binding.pixometer/src/main/java/org/openhab/binding/pixometer/handler/ApiserviceHandler.java

```diff
@@ -0,0 +1,179 @@
+                            - ZonedDateTime.now().toInstant().toEpochMilli();
+                    if (difference <= TOKEN_MIN_DIFF) {
+                        obtainAuthTokenAndExpiryDate(user, password, scope);
+                    }
+
+                } catch (RuntimeException r) {
+                    logger.debug("Could not check token expiry date for Thing {}: ", getThing().getUID(), r);
+                }
+            }
+        }, 1, 1, TimeUnit.DAYS);
+        logger.debug("Refresh job scheduled to run every {} days. for '{}'", 1, getThing().getUID());
+    }
+
+    @Override
+    public void updateStatus(ThingStatus status) {
+        updateStatus(status, ThingStatusDetail.NONE, null);
+    }
+
+    /**
+     * Request auth token with read or write access.
+     * (Write access is prepared for a possible later usage for updating meters.)
+     *
+     * @param user     The username to use
+     * @param password The corresponding password
+     * @param scope    The granted scope on the api for the binding
+     */
+    private void obtainAuthTokenAndExpiryDate(String user, String password, String scope) {
+        try {
+
+            String url = new StringBuilder("https://pixometer.io/api/v1/access-token/").toString();
+            Properties urlHeader = (Properties) new Properties().put("CONTENT-TYPE", "application/json");
+
+            JsonObject httpBody = new JsonObject();
+            httpBody.addProperty("username", user);
+            httpBody.addProperty("password", password);
+            httpBody.addProperty("scope", scope);
+
+            InputStream content = new ByteArrayInputStream(httpBody.toString().getBytes(StandardCharsets.UTF_8));
+            String urlResponse = HttpUtil.executeUrl("POST", url, urlHeader, content, "application/json", 2000);
+            JsonObject responseJson = (JsonObject) new JsonParser().parse(urlResponse);
+
+            if (responseJson.has(CONFIG_BRIDGE_AUTH_TOKEN)) {
+
+                // Store the expire date for automatic token refresh
+                Integer expiresIn = Integer.parseInt(responseJson.get("expires_in").toString());
+                setTokenExpiryDate(ZonedDateTime.now().plusSeconds(expiresIn));
+
+                setAuthToken(responseJson.get(CONFIG_BRIDGE_AUTH_TOKEN).toString().replaceAll("\"", ""));
+                updateStatus(ThingStatus.ONLINE);
+                return;
+            }
+
+            String errorMsg = new StringBuilder("Invalid Api Response ( ").append(responseJson.toString()).append(" )")
+                    .toString();
+
+            throw new Exception(errorMsg);
+
+        } catch (Exception e) {
+            String errorMsg = new StringBuilder("Could not obtain auth token.").append(" ")
+                    .append("Please check your configured account credentials.")
+                    .append(System.getProperty("line.separator")).append(this.getThing().getUID())
```

Comment:
In general you should not put new lines in error log messages. It makes it difficult for systems parsing the logs if it doesn't follow a the convention.
Als this code use benefit from using `String.format`.